### ISSUE TITLE
DB Migration to 6.x may not drop ParentId from posts table

### ISSUE THREAD

**Issue by jsimpso** [mattermost/mattermost-server#20995]
#### Summary
When upgrading from 5.39 to 6.x, if the [medium/high effort](https://docs.mattermost.com/install/self-managed-changelog.html#id134) approach is taken, the `parentid` column will not be dropped from the `posts` table. This will prevent the server from loading any posts.

#### Steps to reproduce
1. Launch a 5.39 instance, make some posts
2. Run through the "medium effort, medium downtime" approach laid out in the [6.0 changelog](https://docs.mattermost.com/install/self-managed-changelog.html#id134), manually executing queries 1-10 and 12-26 from the linked [gist](https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055)
3. Launch the v6 server
4. No posts will load, logs will be presenting errors along the lines of `failed to find Posts with channelId=asdasdasdasdasd: missing destination name parentid in *[]*model.Post"`

#### Expected behavior
At step 3 in the above, I would expect the 6.0 schema migration to execute query 11 (`ALTER TABLE Posts DROP COLUMN ParentId;`).

#### Possible fixes
The workaround is simply to manually execute the missing query once the v6 server has loaded and performed schema migrations.

The issue seems to stem from these lines for mysql and postgresql respectively:

- https://github.com/mattermost/mattermost-server/blob/93174fbe56f3d3dd16b97c84656c4a46305f31cd/db/migrations/mysql/000066_upgrade_posts_v6.0.up.sql#L20
- https://github.com/mattermost/mattermost-server/blob/93174fbe56f3d3dd16b97c84656c4a46305f31cd/db/migrations/postgres/000066_upgrade_posts_v6.0.up.sql#L23

Which appear to have been introduced at PR#19359.

If the guidelines to avoid downtime have been followed, and the other SQL queries have been applied per the gist, then `alter_fileids` and `alert_props` would remain `false` thus preventing the `ParentId` column being dropped from the `posts` table.

**Comment by agnivade** [comment]
You are right @jsimpso. Seems like the `parentid_exist` should have been checked separately. Unfortunately, once these migrations are added, we cannot go back and change them again. We can only add new migrations.

I think we could add a `parentid_exist` check again in our current version as a failsafe, if a user decides to upgrade from 5.39 to 7.4 eventually. Otherwise, they'll have to run manual queries.

cc @isacikgoz @streamer45

**Comment by wgrant** [comment]
The problematic conditional was actually added after the fact, in 56c7da5.

**Comment by isacikgoz** [comment]
Actually we can still add new migration to dot releases with migration number `94` (for now), this wouldn't conflict with future migrations. Or, alternatively, we can add `ALTER TABLE Posts DROP COLUMN ParentId;` to the gist.

**Comment by agnivade** [comment]
`ALTER TABLE Posts DROP COLUMN ParentId;` is already in the gist. But it's suggested to skip that because it's a backwards incompatible change. Adding the new migration to dot releases sounds good to me!

**Comment by hanzei** [comment]
After multiple years of inactivity, we're closing this to help keep our issue tracker focused on active work.
